.gemini/styleguide.md

```diff
@@ -0,0 +1,749 @@
+5. **Include usage examples** using the full symbol location in `keras_hub`.
+6. **Include reference citations** if applicable.
+
+### Example Layer Implementation
+```python
+@keras_hub_export("keras_hub.layers.MyCustomLayer")
+class MyCustomLayer(keras.layers.Layer):
+    """A custom layer for specific functionality.
+
+    This layer implements [specific functionality] as described in
+    [reference paper]. It accepts [input description] and outputs
+    [output description].
+
+    Args:
+        param1: int. Description of parameter 1.
+        param2: str. Description of parameter 2.
+
+    Example:
+    ```python
+    layer = keras_hub.layers.MyCustomLayer(param1=10, param2=20)
+    output = layer(input_tensor)
+    ```
+
+    Reference:
+     - [Author et al., Year](https://arxiv.org/abs/paper_id)
+    """
+
+    def __init__(self, param1, param2, **kwargs):
+        super().__init__(**kwargs)
+        self.param1 = param1
+        self.param2 = param2
+
+    def build(self, input_shape):
+        # Layer building logic
+        super().build(input_shape)
+
+    def call(self, inputs):
+        # Layer computation logic
+        return processed_output
+
+    def get_config(self):
+        config = super().get_config()
+        config.update({
+            "param1": self.param1,
+            "param2": self.param2,
+        })
+        return config
+```
+
+## Checkpoint Conversion
+
+### Script Location
+All checkpoint conversion scripts should be placed in `tools/checkpoint_conversion/`.
+
+### Script Requirements
+- Must be reusable and well-documented
+- Should handle all presets for the model
+- Must demonstrate numerical equivalence with original implementation
+- Should include proper error handling and validation
+
+### Example Conversion Script Structure
+```python
+"""Convert MyModel checkpoints from original format to KerasHub."""
+
+import argparse
+import json
+import os
+
+import keras
+import numpy as np
+
+from keras_hub.src.models.my_model.my_model_backbone import MyModelBackbone
+
+
+def convert_checkpoint(checkpoint_path, output_dir, preset_name):
+    """Convert a MyModel checkpoint to KerasHub format."""
+    # Load original checkpoint
+    # Convert weights to KerasHub format
+    # Save in KerasHub format
+    # Validate numerical equivalence
+    pass
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Convert MyModel checkpoints")
+    parser.add_argument("--checkpoint_path", required=True)
+    parser.add_argument("--output_dir", required=True)
+    parser.add_argument("--preset_name", required=True)
+    
+    args = parser.parse_args()
+    convert_checkpoint(args.checkpoint_path, args.output_dir, args.preset_name)
+
+
+if __name__ == "__main__":
+    main()
+```
+
+## HuggingFace Converters
+
+### When to Add
+If the model is being ported from HuggingFace, a converter must be added to `keras_hub/src/utils/transformers/`.
```

Comment:
`keras_hub/src/utils/transformers/` for transformer based model and `keras_hub/src/utils/timm/` for PyTorch image models.